src/lib/nvme_pt_ioctl.c

```diff
@@ -2311,3 +2320,70 @@ int opal_write_shadow_mbr_pt(struct sed_device *dev, const struct sed_key *key,
 	opal_end_session(dev->fd, opal_dev);
 	return ret;
 }
+
+
+int opal_revertsp_pt(struct sed_device * dev, const struct sed_key *key, bool keep_key)
+
+{
+	
+	int ret;
+	struct opal_device * opal_dev;
+
+	if (key == NULL || dev == NULL) {
+		SEDCLI_DEBUG_MSG("User must provide a valid pointer\n");
+		return -EINVAL;
+	}
+
+	opal_dev = dev->priv;
+
+	ret = opal_start_admin1_lsp_session(dev->fd, opal_dev, key);
+
+	if (ret) {
+		opal_end_session(dev->fd,opal_dev);
+		return ret;
+	}
+	
+		
+	ret = opal_revertlsp_local(dev->fd, opal_dev, keep_key);
+	
+	opal_end_session(dev->fd,opal_dev);
+	
+	return ret;
+
+
+}
+
+
+int opal_non_destructive_revert_pt(struct sed_device *dev, const struct sed_key *key)
```

Comment:
This additional function seems repetitive. I believe this isn't required.

Reply:
Im confused. This is the function that calls the function above it and does the non destructive revert? If I remove this function im confused on how the non destructive revert will work?